examples/modelzoo/evaluate_caffe_net.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+"""Example code of evaluating a Caffe reference model for ILSVRC2012 task.
+
+Prerequisite: To run this example, crop the center of ILSVRC2012 validation
+images and scale them to 256x256, and make a list of space-separated CSV each
+column of which contains a full path to an image at the fist column and a zero-
+origin label at the second column (this format is same as that used by Caffe's
+ImageDataLayer).
+
+"""
+from __future__ import print_function
+import argparse
+import sys
+
+import cv2
+import numpy as np
+
+import chainer
+from chainer import cuda
+import chainer.functions as F
+from chainer.functions import caffe
+
+
+parser = argparse.ArgumentParser(
+    description='Evaluate a Caffe reference model on ILSVRC2012 dataset')
+parser.add_argument('dataset', help='Path to validation image-label list file')
+parser.add_argument('model_type', choices=('alexnet', 'caffenet', 'googlenet'),
+                    help='Model type (alexnet, caffenet, googlenet)')
+parser.add_argument('model', help='Path to the pretrained Caffe model')
+parser.add_argument('--mean', '-m', default='ilsvrc_2012_mean.npy',
+                    help='Path to the mean file')
+parser.add_argument('--batchsize', '-B', type=int, default=100,
+                    help='Minibatch size')
+parser.add_argument('--gpu', '-g', type=int, default=-1,
+                    help='Zero-origin GPU ID (nevative value indicates CPU)')
+args = parser.parse_args()
+assert args.batchsize > 0
+
+
+dataset = []
+with open(args.dataset) as list_file:
+    for line in list_file:
+        pair = line.strip().split()
+        dataset.append((pair[0], np.int32(pair[1])))
+
+assert len(dataset) % args.batchsize == 0
+
+
+func = caffe.CaffeFunction(args.model)
```

Comment:
Loading protobuf file is very slow. Showing log message is helpful for users such as "Opening caffe model file...".